src/katgpucbf/xbgpu/engine.py

```diff
@@ -794,9 +938,13 @@ def __init__(
         # time. This keeps the pipelines synchronised to avoid running out of
         # RxQueueItems.
         self._active_in_sem = asyncio.BoundedSemaphore(1)
-        self._pipelines = [
-            XPipeline(output, self, context, tx_enabled) for output in outputs if isinstance(output, XOutput)
-        ]
+
+        self._pipelines: list[Pipeline] = []
+        x_outputs = [output for output in outputs if isinstance(output, XOutput)]
+        b_outputs = [output for output in outputs if isinstance(output, BOutput)]
+        self._pipelines = [XPipeline(x_output, self, context, tx_enabled) for x_output in x_outputs]
+        if len(b_outputs):
```

Comment:
I think you can just do 
```suggestion
        if b_outputs:
```